### DISCUSSION TITLE
How to get started? How to create the first Admin User?

### DISCUSSION THREAD

**Discussion by mojibake-umd** [UMN-LATIS/z#144] [Q&A] [answered]
First let me say thank you for providing code on github, it looks like a wonderful project.  
I work with UMD (University of Maryland) and was looking to evaluate Z as replacement for our aging URL Shortner service.
Following the ReadMe I was able to somewhat past the starting line.
However the documentation does not seem to indicate what to do once the app is up and running.
I have a feeling that my LDAP config may not be correct, as clicking on Sign-In asks to provide Name and Email...
Only using sqlite to test things out.
I have searched around the config directory and code, but can't seem to find how to add the first Admin user.
The sqlite database does not seem to provide any table/info that would indicate or allow me to insert a user, as I do not see an admin table, or anything indicating how Admin users are defined.

Any information would be appreciated.

**Comment by jxjj** [comment, marked as answer]
Thanks for the kind words!

To stand up Z locally without UMN config, you'll need to modify the setup instructions a bit. I'm guessing you were seeing a few LDAP bind errors.

First, you'll need to skip the seeding since that does a user lookup via ldap. So instead of `rails db:setup` do:
```
$ ./bin/rails db:create
$ ./bin/rails db:schema:load
```

Then, we can tell the app to use a mock LDAP User Lookup service at startup (like we do we running CI tests) by passing an environment variable:

```
$ USER_LOOKUP_SKELETON=1 bin/dev
```

(Sorry, this isn't well documented – we need to surface it better.)

Once things are started, you can login locally with whatever credentials (e.g. `admin`, `admin`). Z will create a user with an `admin` uid in the `users` table. You can promote the user to `admin` by changing the value of their `admin` column from null to 1.

Alternately, you could use the rails console for promotion. Use `bin/rails c` to start the rails console, then:

```ruby
my_admin = User.find_by(uid: "admin")
my_admin.admin = true
my_admin.save()
```

Here's a short video walk through using sqlite and the ldap skipping if it helps:
[<img src="https://user-images.githubusercontent.com/980170/236949830-3fdd10b5-fa8e-416e-b0e0-bad265490b6e.png" width="640" alt="walkthrough" />](https://youtu.be/yyJ5YqbsTYQ)

**Reply by mojibake-umd** [reply]
Thank you very much Jason, that was just that extra bit I needed to be able to explore the functionality of the Admin role, and the application as a whole.
